flutter/shell/platform/tizen/tizen_window_ecore_wl2.cc

```diff
@@ -47,14 +53,107 @@ FlutterPointerDeviceKind ToFlutterDeviceKind(const Ecore_Device* dev) {
   }
 }
 
+#ifdef TV_PROFILE
+time_t GetBootTimeEpoch() {
+  struct timespec now, boot_time;
+  if (clock_gettime(CLOCK_REALTIME, &now) != 0) {
+    FT_LOG(Error) << "Fail to get clock_gettime(CLOCK_REALTIME).";
+    return -1;
+  }
+  if (clock_gettime(CLOCK_BOOTTIME, &boot_time) != 0) {
+    FT_LOG(Error) << "Fail to get clock_gettime(CLOCK_BOOTTIME).";
+    return -1;
+  }
+  time_t boot_time_epoch = now.tv_sec - boot_time.tv_sec;
+  return (boot_time_epoch / 10) * 10;
+}
+
+bool GetPointingDeviceToastPreference() {
+  bool show_unsupported_toast = false;
+  time_t boot_time = GetBootTimeEpoch();
+
+  if (boot_time != -1) {
+    char* app_id = nullptr;
+    int ret = app_get_id(&app_id);
+    if (ret != APP_CONTROL_ERROR_NONE || !app_id) {
+      FT_LOG(Error) << "Fail to get app id.";
+      return false;
+    }
+
+    std::ostringstream boot_time_buffer;
```

Comment:
Line 73 ~ 87, can we extract a new method for preference_key?

Reply:
I added GetPreferenceKey() method.